tests/transforms/test_generate_heatmapd.py

```diff
@@ -0,0 +1,267 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import unittest
+
+import numpy as np
+import torch
+from parameterized import parameterized
+
+from monai.data import MetaTensor
+from monai.transforms.post.dictionary import GenerateHeatmapd
+from tests.test_utils import assert_allclose
+
+# Test cases for dictionary transforms with reference image
+# Only test with non-MetaTensor types to avoid affine conflicts
+TEST_CASES_WITH_REF = []
+TEST_CASES_WITH_REF.append(
+    [
+        "dict_with_ref_3d_numpy",
+        np.array([[2.5, 2.5, 3.0], [5.0, 5.0, 4.0]], dtype=np.float32),
+        {"sigma": 2.0},
+        (2, 8, 8, 8),
+        torch.float32,
+        True,  # uses reference image
+    ]
+)
+TEST_CASES_WITH_REF.append(
+    [
+        "dict_with_ref_3d_torch",
+        torch.tensor([[2.5, 2.5, 3.0], [5.0, 5.0, 4.0]], dtype=torch.float32),
+        {"sigma": 2.0},
+        (2, 8, 8, 8),
+        torch.float32,
+        True,  # uses reference image
+    ]
+)
```

Comment:
```suggestion
TEST_CASES_WITH_REF = [
    [
        "dict_with_ref_3d_numpy",
        np.array([[2.5, 2.5, 3.0], [5.0, 5.0, 4.0]], dtype=np.float32),
        {"sigma": 2.0},
        (2, 8, 8, 8),
        torch.float32,
        True,  # uses reference image
    ],
    [
        "dict_with_ref_3d_torch",
        torch.tensor([[2.5, 2.5, 3.0], [5.0, 5.0, 4.0]], dtype=torch.float32),
        {"sigma": 2.0},
        (2, 8, 8, 8),
        torch.float32,
        True,  # uses reference image
    ]
]
```
I think it's fine to define lists directly like this.